client/src/document.js

```diff
@@ -133,47 +133,74 @@ function SidebarLeaf({ depth, title, content }) {
-  sections.push(<Attributes key="attributes" document={document} />);
-  sections.push(
-    <ProseWithHeading key="usage_notes" section={document.prose.usage_notes} />
-  );
-  sections.push(
-    <ProseWithHeading
-      key="accessibility_concerns"
-      section={document.prose.accessibility_concerns}
-    />
-  );
-  sections.push(<Examples key="examples" document={document} />);
-  sections.push(
-    <BrowserCompatibility key="browser_compatibility" document={document} />
-  );
-  sections.push(
-    <ProseWithHeading key="see_also" section={document.prose.see_also} />
-  );
+function RenderDocumentBody({ document }) {
+  const sections = [];
+  /**
+   * The reason we can't return a filtered map of 'document.body' is because
+   * some of the parts of 'document.body' is an array.
+   * E.g. document.body.additional_prose == [
+   *     {type: 'prose', value: STUFF},
+   *     {type: 'prose', value: OTHER_STUFF},
+   * ]
+   */
+  document.body.forEach((section, i) => {
+    if (section.type === "prose") {
+      sections.push(<Prose key={section.value.id} section={section.value} />);
+    } else if (section.type === "additional_prose") {
+      section.value.forEach((subsection, j) => {
+        if (subsection.type === "prose" && subsection.value) {
+          sections.push(
+            <ProseWithHeading
+              key={`${subsection.title}${i}${j}`}
+              section={subsection.value}
+            />
+          );
+        } else {
+          console.warn("Don't know how to deal with subsection:", subsection);
+        }
+      });
+    } else if (section.type === "interactive_example") {
+      sections.push(
+        <InteractiveExample
+          key={section.value}
+          src={section.value}
+          document={document}
+        />
+      );
+    } else if (section.type === "attributes") {
+      sections.push(
+        <Attributes key={`attributes${i}`} attributes={section.value} />
+      );
+    } else if (section.type === "browser_compatibility") {
+      sections.push(
+        <BrowserCompatibility
+          key="browser_compatibility"
+          content={section.value}
+        />
+      );
+    } else if (section.type === "examples") {
+      sections.push(<Examples key={`examples${i}`} examples={section.value} />);
+    } else if (section.type === "info_box") {
+      // XXX Unfinished!
+      // https://github.com/mdn/stumptown-content/issues/106
+      console.warn("Don't know how to deal with info_box!");
+      // console.log(section);
+    } else {
+      console.log("???", section.type, section);
+    }
+  });
 
   return sections;
 }
 
 function Prose({ section }) {
-  if (!section) {
-    return null;
-  }
   return <div dangerouslySetInnerHTML={{ __html: section.content }} />;
 }
 
 function ProseWithHeading({ id, section }) {
-  if (!section) {
-    return null;
+  if (!id) {
+    // XXX Is this safe/right?
+    id = section.title.replace(/\s+/g, "_").trim();
```

Comment:
Seems reasonable to me

Reply:
The `id` here isn't specifically intended to be used to provide `id` attributes. It's intended to enable some other consumer of the stumptown-content JSON to retrieve a specific prose section (e.g. to get the short description).

It's OK to use it for `id` attributes, but I think it would be worth thinking about questions like why we want to set `id` attributes on these elements, which elements we want to set them on, and whether we want the value to be the same for different translations of the same page.

For example:

* the renderer in this PR is only setting the attribute for H2 headings attached to prose sections, not for other headings or or H2 headings attached to other page elements like lists of attributes or BCD tables. Is that the intention?

* This PR is sometimes using the `id` value from the JSON (which is probably supposed to be common for different translations of the same item), but can sometimes derive the value from the title text (which of course will be different for different translations) . Is that OK, given the use cases for this attribute?

Reply:
I'm still confused. 

I made [the fix](https://github.com/mdn/stumptown-renderer/pull/89/commits/128eebc4e5317fa9d925ba58dd913e6dae15f190) so that `short_description` and `overview` is rendered without a heading. 

So, if a section doesn't have an `id`, should it then omit the `id="..."` attribute and just display `<h2>{section.title}</h2>` alone?